shell/key-bindings.fish

```diff
@@ -69,8 +69,13 @@ function fzf_key_bindings
       # history's -z flag was added in fish 2.4.0, so don't use it for versions
       # before 2.4.0.
       if [ "$FISH_MAJOR" -gt 2 -o \( "$FISH_MAJOR" -eq 2 -a "$FISH_MINOR" -ge 4 \) ];
-        history -z | eval (__fzfcmd) --read0 --print0 -q '(commandline)' | read -lz result
-        and commandline -- $result
+        if type -q perl
+          history -z --reverse | perl -0 -pe 's/^/$.\t/g; s/\n/\n\t/gm' | eval (__fzfcmd) --tac --read0 --print0 -q '(commandline)' | sed 's/^[0-9]*\t//' | read -lz result
```

Comment:
> New `fish` version using perl.



```suggestion
          history -z --reverse | perl -0 -pe 's/^/$.\t/g; s/\n/\n\t/gm' | eval (__fzfcmd) --delimiter '\t' --nth 2..,.. --tac --read0 --print0 -q '(commandline)' | sed 's/^[0-9]*\t//' | read -lz result
```

Reply:
Thanks for the changes to `fish`. 

I worked on the `zsh` version, but had to use `perl` as well. Will test it for a day and make a PR.

```sh
printf '%1$s\t%2$s\000' "${(vk)history[@]}" | 
  perl -0 -ne 'if (!$seen{(/^\s*[0-9]+\**\s+(.*)/, $1)}++) { s/\n/\n\t/gm; print; }' |
  fzf --read0
```

Reply:
> USE OF '\t' IN SED SCRIPTS: For clarity in documentation, we have used
the expression '\t' to indicate a tab character (0x09) in the scripts.
However, most versions of sed do not recognize the '\t' abbreviation,
so when typing these scripts from the command line, you should press
the TAB key instead. '\t' is supported as a regular expression
metacharacter in awk, perl, and in a few implementations of sed.
> 
> https://www.unixguide.net/unix/sedoneliner.shtml

In my version of `sed`, the `\t` character is not recognized. However, `[[:blank:]]` or `[[:space:]]` works with the `-E` flag, or a literal tab character. The latter option is not my preferred choice because the tab character is invisible in scripts and could be accidentally altered when copying and pasting.


`fish` does not support  C-Quotes `$'\t'` either

<img src="https://github.com/junegunn/fzf/assets/92653266/ab8a2d1a-901b-4193-a2b0-479a7319e5ae" width="400">

I would suggest drop `sed` for a `perl` or `awk` solution.

```sh
perl -pe 's/^\d+\t//'
```

Reply:
I think `s/\n/\n\t/gm` makes the history look better in `fzf` but the result is an additional `tab` char, i think we should not alter the command or revert the extra tab `'s/\n\t/\n/gm'`?

Reply:
I used `[0-9]*` instead of `+` to strip leading tabs in every line.

Reply:
> I would suggest drop `sed` for a `perl` or `awk` solution.

Sure, see 3afd543 and 700bf1d

Reply:
Thank you. I'm unclear on why `--nth 2..,..` wasn't included as well.

Reply:
It was an oversight. Now fish also needs it.

Reply:
21f8d88f18a7327d67986f6a534b7b087e6c7fdc